sdk/openai/Azure.AI.OpenAI/tests/AzureChatExtensionsTests.cs

```diff
@@ -0,0 +1,152 @@
+{
+    public AzureChatExtensionsTests(bool isAsync)
+        : base(isAsync)//, RecordedTestMode.Live)
+    {
+    }
+
+    public enum ExtensionObjectStrategy
+    {
+        WithGenericParentType,
+        WithScenarioSpecificHelperType
+    }
+
+    [RecordedTest]
+    [TestCase(OpenAIClientServiceTarget.Azure, ExtensionObjectStrategy.WithGenericParentType)]
+    [TestCase(OpenAIClientServiceTarget.Azure, ExtensionObjectStrategy.WithScenarioSpecificHelperType)]
+    public async Task BasicSearchExtensionWorks(
+        OpenAIClientServiceTarget serviceTarget,
+        ExtensionObjectStrategy extensionStrategy)
+    {
+        OpenAIClient client = GetTestClient(serviceTarget);
+        string deploymentOrModelName = OpenAITestBase.GetDeploymentOrModelName(
+            serviceTarget,
+            OpenAIClientScenario.ChatCompletions);
+
+        AzureChatExtensionsOptions extensionsOptions = new();
+        extensionsOptions.Extensions.Add(extensionStrategy switch
+        {
+            ExtensionObjectStrategy.WithGenericParentType => new AzureChatExtensionConfiguration()
+            {
+                Type = "AzureCognitiveSearch",
+                Parameters = BinaryData.FromObjectAsJson(new
+                {
+                    Endpoint = "https://openaisdktestsearch.search.windows.net",
+                    IndexName = "openai-test-index-carbon-wiki",
+                    Key = GetCognitiveSearchApiKey().Key,
+                },
+                new JsonSerializerOptions() { PropertyNamingPolicy = JsonNamingPolicy.CamelCase }),
+            },
+            ExtensionObjectStrategy.WithScenarioSpecificHelperType
+                => new AzureCognitiveSearchChatExtensionConfiguration()
+                {
+                    SearchEndpoint = new Uri("https://openaisdktestsearch.search.windows.net"),
+                    IndexName = "openai-test-index-carbon-wiki",
+                    SearchKey = GetCognitiveSearchApiKey(),
+                },
+            _ => throw new NotImplementedException("Don't know how to add the extension config!"),
+        });
+
+        var requestOptions = new ChatCompletionsOptions()
+        {
+            Messages =
+            {
+                new ChatMessage(ChatRole.User, "What does PR complete mean?"),
+            },
+            MaxTokens = 512,
+            AzureExtensionsOptions = extensionsOptions,
+        };
+
+        Response<ChatCompletions> response = await client.GetChatCompletionsAsync(deploymentOrModelName, requestOptions);
+        Assert.That(response, Is.Not.Null);
+        Assert.That(response.Value, Is.Not.Null);
+        Assert.That(response.Value.Choices, Is.Not.Null.Or.Empty);
+
+        ChatChoice firstChoice = response.Value.Choices[0];
+        Assert.That(firstChoice, Is.Not.Null);
+        Assert.That(firstChoice.FinishReason, Is.EqualTo(CompletionsFinishReason.Stopped));
+        Assert.That(firstChoice.Message.Role, Is.EqualTo(ChatRole.Assistant));
```

Comment:
Given that we can record service responses, and save them, we should be able to do proper validation of returned responses in a consistent way

Reply:
Right. This would be questionable for live service testing, but given we're working from recordings it *should* remain consistent.

The most important thing about it is that it's validating that expected values get deserialized; we've had at least one case where *not* having checks like this missed that we weren't deserializing from the right field (and thus weren't changing one of the result fields to the non-default value it should have been).